### ISSUE TITLE
"Something went wrong while linking"

### ISSUE THREAD

**Issue by gschattenhofer** [airbnb/lottie-react-native#89]
When I'm trying to link i'm getting some errors.

```
react-native link lottie-ios
Linking lottie-ios ios dependency 
It seems something went wrong while linking. Error: buildSettings.LIBRARY_SEARCH_PATHS.push is not a function 
```

```

react-native link lottie-react-native
DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
Linking lottie-react-native android dependency 
Android module lottie-react-native has been successfully linked 
Linking lottie-react-native ios dependency 
ERR! It seems something went wrong while linking. Error: buildSettings.LIBRARY_SEARCH_PATHS.push is not a function 
```

react-native-cli: 2.0.1
react-native: 0.37.0
node -v v7.5.0

I'll be trying CocoaPods now, but i'm interested on why i'm getting that error.
=== Edit
I've tried adding lottie on Podfile (also added react on it, cause i was getting a deprecated message). Run pod install successfully but i was not able to reproduce an animation. I've used Lottie Logo json as an example. I've tried the link commands above after running pod install, and returned for lottie-ios "iOS module lottie-ios is already linked "

But for lottie-react-native still got:
Linking lottie-react-native ios dependency 
It seems something went wrong while linking. Error: buildSettings.LIBRARY_SEARCH_PATHS.push is not a function

**Comment by g3r4n** [comment]
do not use `react-native link lottie-ios` see PR #86 to link manually the lottie iOS framework

**Comment by gschattenhofer** [comment]
I've linked manually tha lottie as it's on that PR. I see the Lottie.framework in build/Debug-iphoneos on Embedded Binaries. Still can not manage to play the animation on iOS. If the verification added on that PR is having the framework on that list, then i guess this issue no longer a Linking problem. I've verified already that LibLottie.a is not under Linked Frameworks and Libraries, but i do have, ofcourse Lottie.framework on it.

**Comment by g3r4n** [comment]
Have you rebuild the project ? Can you share a repository in order to reproduce the error ?

**Comment by g3r4n** [comment]
Here you can find a working example : 
https://github.com/g3r4n/react-native-lottie-example

**Comment by joypatel04** [comment]
I have been facing this problem from a week. @g3r4n  I want to integrate library into my project i check the above example and it's working well but when i am installing lottie-react-native and perform necessary steps it works well but on our CI it throws and error linker comment failed.

**Comment by gschattenhofer** [comment]
I've manage to solve my problem. On the PR that @g3r4n posted above tells you how to manually link lottie-ios on xCode. I've checked the project he also sent, react-native-lottie-example and realize that you also have to manually link lottie-react-native: On right-click on Libraries, "Add files to...", select the xcode project file inside lottie-react-native on your node_modules. After that on "Linked frameworks and Libraries" you should add only the libLottieReactNative.a from LottieReactNative.xcodeproj . Then clean and rebuild. Make sure you dont have Lottie on the pods, thats not necessary. Now if you run react-native link lottie-react-native, should tell you that is already linked on ios. And you should be able to see the animation playing.

So, in other words, forget adding it on podfile and react-native link will not work for lottie-ios nor lottie-react-native, you have to linke them manually.

**Comment by joypatel04** [comment]
@gschattenhofer It's working on my local Machine but it's failing on Team City Continuos integration 

[12:15:12][Ld] ld: library not found for -lLottieReactNative
[12:15:12][Ld] clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by gschattenhofer** [comment]
I'm unexperienced with that. Sorry i can't help you.

**Comment by joypatel04** [comment]
I did all the steps @g3r4n explained and it's working on simulator but when i commit it and it goes to CI Server it's failing with above ld: library not found for -lLottieReactNative/linker command failed

Embedded Library
------------------- 
<img width="539" alt="screen shot 2017-03-14 at 12 22 38 pm" src="https://cloud.githubusercontent.com/assets/18585502/23918652/11f2ef0a-08b1-11e7-95c3-837db19f0f33.png">

Linked framework and Library
------------------------------
<img width="549" alt="screen shot 2017-03-14 at 12 22 02 pm" src="https://cloud.githubusercontent.com/assets/18585502/23918633/054f133c-08b1-11e7-9472-95b7bb914dd2.png">

Manually Added Lottie-ios & lottie-react-native 
------------------------------------------------
<img width="251" alt="screen shot 2017-03-14 at 12 22 26 pm" src="https://cloud.githubusercontent.com/assets/18585502/23918660/16862c62-08b1-11e7-98f6-3791495eb088.png">

**Comment by joypatel04** [comment]
@lelandrichardson Do you have any idea that why it's failing on CI and not failing on Local Dev Machine?

**Comment by joypatel04** [comment]
Looks like issue was regarding Schemes that i have so i figured nothing wrong with @g3r4n Setup step that's how it suppose to be. And even if any one seen error regarding libLottieReactNative not found something just remove it from Link Framework and Library and add again.

**Comment by g3r4n** [comment]
@joypatel04

**Comment by lelandrichardson** [comment]
I believe this should now be fixed in `lottie-react-native@1.1.1`. Please try it out and let me know if you still have trouble. Closing this issue for now. Thanks!

**Comment by ShariqMush** [comment]
I'm having trouble with doing `react-native-link`, gives me the same error
```
rnpm-install info Linking lottie-react-native android dependency
rnpm-install info Android module lottie-react-native has been successfully linked
rnpm-install info iOS module lottie-react-native is already linked
rnpm-install ERR! Something went wrong while linking. Error: spawn UNKNOWN
Please file an issue here: https://github.com/facebook/react-native/issues

spawn UNKNOWN
```
I'm developing for Android on Windows. i have tried doing it on several other computers but still the same issue remains.

**Comment by skovhus** [comment]
React native link seems to assume that `LIBRARY_SEARCH_PATHS` is a list.

So inside `project.pbxproj` change
`LIBRARY_SEARCH_PATHS = "$(BUILD...";`

to 
`LIBRARY_SEARCH_PATHS = ("$(BUILD...");`

**Comment by nazwa** [comment]
@joypatel04 I know it's been ages, but do you remember what was your Schemes issue and how did you solve it?
